backend/inspirehep/pidstore/models.py

```diff
@@ -0,0 +1,139 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2019 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+import datetime
+
+import structlog
+from invenio_db import db
+from invenio_pidstore.errors import PIDInvalidAction
+from invenio_pidstore.models import PIDStatus
+from sqlalchemy.exc import IntegrityError, SQLAlchemyError
+from sqlalchemy_utils.models import Timestamp
+
+from inspirehep.pidstore.errors import (
+    PidRedirectionMissing,
+    PidStatusBroken,
+    WrongPidTypeRedirection,
+    WrongRedirectionPidStatus,
+)
+
+LOGGER = structlog.getLogger()
+
+
+class InspireRedirect(db.Model, Timestamp):
+    __tablename__ = "inspire_pidstore_redirect"
+
+    id = db.Column(db.Integer, primary_key=True)
+    original_pid_id = db.Column(
+        db.Integer,
+        db.ForeignKey(
+            "pidstore_pid.id", name="fk_inspire_pidstore_redirect_old_pid_id"
+        ),
+        nullable=False,
+        index=True,
+        unique=True,
+    )
+
+    original_pid = db.relationship(
+        "PersistentIdentifier",
+        backref="redirection",
+        foreign_keys=original_pid_id,
+        innerjoin=True,
+        uselist=False,
+    )
+    new_pid_id = db.Column(
+        db.Integer,
+        db.ForeignKey(
+            "pidstore_pid.id", name="fk_inspire_pidstore_redirect_new_pid_id"
+        ),
+        nullable=False,
+        index=True,
+    )
+
+    new_pid = db.relationship(
+        "PersistentIdentifier",
+        backref="redirected_pids",
+        foreign_keys=new_pid_id,
+        innerjoin=True,
+        uselist=False,
+    )
+
+    @classmethod
+    def _fix_existing_redirection(cls, redirection, old_pid, new_pid):
```

Comment:
```suggestion
    def _change_existing_redirection(cls, redirection, old_pid, new_pid):
```
I was confused, but IIUC, this is not fixing the redirection so much as changing which record it should redirect to.

Reply:
:+1: